redisinsight/ui/src/pages/browser/modules/key-details/components/rejson-details/rejson-object/RejsonObject.tsx

```diff
@@ -1,209 +1,108 @@
   const [loading, setLoading] = useState<boolean>(false)
-  const [error, setError] = useState<string | null>(null)
+  const [isExpanded, setIsExpanded] = useState<boolean>(expadedRows?.has(path) || false)
 
   useEffect(() => {
-    setError(null)
-  }, [newKey, newValue, valueOfEntireObject])
-
-  const handleFormSubmit = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    onClickAddNewKVPair()
-  }
-
-  const handleResetAddNewKeyValuePair = () => {
-    setError(null)
-    setAddNewKeyValuePair(false)
-    setNewKey('')
-    setNewValue('')
-  }
-
-  const handleUpdateValueFormSubmit = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault()
-    setEditEntireObject(false)
-    onClickSubmitEntireObject()
-  }
-
-  const handleOnEsc = (e: KeyboardEvent, type: string) => {
-    if (e.code?.toLowerCase() === 'escape' || e.keyCode === 27) {
-      e.stopPropagation()
-      setError(null)
-
-      if (type === 'add') handleResetAddNewKeyValuePair()
-      if (type === 'edit') setEditEntireObject(false)
-    }
-  }
-
-  const onClickEditEntireObject = () => {
-    setEditEntireObject(!editEntireObject)
-
-    if (!editEntireObject) {
-      handleFetchVisualisationResults(path, true).then((data: REJSONResponse) => {
-        setValueOfEntireObject(typeof data.data === 'object' ? JSON.stringify(data.data, undefined, 4) : data.data)
-      })
-    }
-  }
-
-  const onClickSubmitEntireObject = () => {
-    setEditEntireObject(false)
-    const error: string | null = validateRejsonValue(valueOfEntireObject)
-
-    if (error) {
-      setError(error)
-      setEditEntireObject(true)
+    if (!isExpanded) {
+      setValue(defaultValue)
       return
     }
 
-    const body: IJSONObject = {
-      path,
-      value: valueOfEntireObject as JSONScalarValue,
-      operation: 'update'
+    if (isDownloaded) {
+      setValue(currentValue)
+    } else {
+      fetchObject()
     }
+  }, [isExpanded, isDownloaded])
 
-    handleSubmitJsonUpdateValue(body)
-    handleSetRejsonDataAction(selectedKey, path, valueOfEntireObject as string)
-  }
-
-  const onClickSetKVPair = () => {
-    setAddNewKeyValuePair(!addNewKeyValuePair)
-    setNewKey('')
-    setNewValue('')
-  }
-
-  const onClickAddNewKVPair = () => {
+  const handleFormSubmit = ({ key, value }: { key?: string, value: string }) => {
     setAddNewKeyValuePair(false)
 
-    const error: string | null = validateKeyValue()
-    if (error) {
-      setAddNewKeyValuePair(true)
-      setError(error)
+    if (type === ObjectTypes.Array) {
+      handleAppendRejsonObjectItemAction(selectedKey, path, value)
       return
     }
 
-    const body: IJSONObject = {
-      previous_path: path,
-      new_key: newKey,
-      path: `${path}[${newKey}]`,
-      operation: 'update',
-      value: newValue
+    try {
+      const unescapedKey = JSON.parse(key as string)
+      const updatedPath = unescapedKey.includes('"') ? `${path}['${unescapedKey}']` : `${path}["${unescapedKey}"]`
```

Comment:
you have almost the same logic in rejesonDetails, let move this logic to some util, like `getUnescapedKey(key)`